cmake/Platform/Common/Install_common.cmake

```diff
@@ -525,15 +525,20 @@ function(ly_setup_runtime_dependencies)
     if(COMMAND ly_setup_runtime_dependencies_copy_function_override)
         ly_setup_runtime_dependencies_copy_function_override()
     else()
-        ly_install(CODE
+        # despite this copy function being the same, we need to install it per component that uses it
+        # (which is per-configuration per-permutation component)
+        foreach(conf IN LISTS CMAKE_CONFIGURATION_TYPES)
+            string(TOUPPER ${conf} UCONF)
+            ly_install(CODE
 "function(ly_copy source_file target_directory)
     cmake_path(GET source_file FILENAME file_name)
     if(NOT EXISTS \${target_directory}/\${file_name})
         file(COPY \"\${source_file}\" DESTINATION \"\${target_directory}\" FILE_PERMISSIONS ${LY_COPY_PERMISSIONS})
     endif()
 endfunction()"
-            COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
-        )
+                COMPONENT ${LY_INSTALL_PERMUTATION_COMPONENT}_${UCONF}
```

Comment:
Just to confirm, nothing in the default component uses this function, right?

Reply:
right, this copy function is used by the code bellow (copy commands are gathered 568-574 and the code is installed in 580-585).

Under some circumstances, it seemed that installing the function in the core component was being visible to all the other components, but seems sometimes cpack calls install with individual components, and since we have a different component for each configuration, the function may not exist (which silently fails in cpack)

This ends up with multiple copies of the `ly_copy` function in the `cmake_install.cmake` file, but all install calls always add a `if("x${CMAKE_INSTALL_COMPONENT}x" STREQUAL` line. We could simplify this in the future by appending to the cmake_install.cmake file directly.